rest-api-spec/src/main/resources/rest-api-spec/test/indices.delete_index_template/10_basic.yml

```diff
@@ -0,0 +1,52 @@
+setup:
+  - do:
+      indices.put_index_template:
+        name: test_template_1
+        body:
+          index_patterns: test-*
+          template:
+            settings:
+              number_of_shards:   1
+              number_of_replicas: 0
+          "priority": 50
+
+  - do:
+      indices.put_index_template:
+        name: test_template_2
+        body:
+          index_patterns: test-*
+          data_stream: {}
+          template:
+            settings:
+              number_of_shards:   1
+              number_of_replicas: 0
+          "priority": 51
+
+---
+teardown:
+  - do:
+      indices.delete_data_stream:
+        name: test-1
+        ignore: 404
+  - do:
+      indices.delete_index_template:
+        name: test_template_1
+        ignore: 404
+  - do:
+      indices.delete_index_template:
+        name: test_template_2
+        ignore: 404
+
+---
+"Delete index template which is not used by data stream but index pattern matches":
+  - skip:
+      version: " - 2.16.99"
```

Comment:
We may need to change it to `2.99.99` for `main` and update after backport, giving the check one more chance